File: hseolymp/2016-17/demo/mars.cpp
#include <cmath>
#include <iostream>

using namespace std;

int main()
{
	double n;
	cin >> n;
	int days = 0;
	double factory = 1.0;

	int without_fact = ceil((factory + sqrt(factory * factory + 8 * n * factory)) / (2 * factory));
	int with_fact = ceil((2 * factory + sqrt(4 * factory * factory + 16 * n * factory)) / (4 * factory) + 1);
	while (with_fact < without_fact)
	{
		factory *= 2;
		++days;
		without_fact = ceil((factory + sqrt(factory * factory + 8 * n * factory)) / (2 * factory));
		with_fact = ceil((2 * factory + sqrt(4 * factory * factory + 16 * n * factory)) / (4 * factory) + 1);
	}
	days += ceil((factory + sqrt(factory * factory + 8 * n * factory)) / (2 * factory));
	cout << days << endl;
	return 0;
}

File: hseolymp/2016-17/demo/mars.py
import math
n = int(input())

factory = 1
days = 0

without_fact = int(round(0.5 + (factory + math.sqrt(factory**2 + 8*n*factory)) / (2*factory)))
with_fact = int(round(0.5 + (2*factory + math.sqrt((2*factory)**2 + 8*n*2*factory)) / (2*2*factory))) + 1

while with_fact < without_fact:
    factory *= 2
    days += 1
    without_fact = int(round(0.5 + (factory + math.sqrt(factory**2 + 8*n*factory)) / (2*factory)))
    with_fact = int(round(0.5 + (2*factory + math.sqrt((2*factory)**2 + 8*n*2*factory)) / (2*2*factory))) + 1

days += int(round(0.5 + (factory + math.sqrt(factory**2 + 8*n*factory)) / (2*factory)))
print(days)
